dowhy/causal_refuters/bootstrap_refuter.py

```diff
@@ -1,184 +1,208 @@
+    
+    Parameters
+    -----------
+    -'num_simulations': int, CausalRefuter.DEFAULT_NUM_SIMULATIONS by default
+    The number of simulations to be run
+    - 'sample_size': int, Size of the original data by default
+    The size of each bootstrap sample
+    - 'required_variables': int, list
+    A user can input either an integer value or a list.
+        1. An integer argument refers to how many confounders  will be modified
+        2. A list allows the user to explicitly refer to which confounders should be seleted to be made noisy
+    - 'noise': float, BootstrapRefuter.DEFAULT_STD_DEV by default
+    The standard deviation of the noise to be added to the data
+    - 'probability_of_change': float, 'noise' by default if the value is less than 1
+    It specifies the probability with which we change the data for a boolean or categorical variable
+    - 'random_state': int, RandomState, None by default
+    The seed value to be added if we wish to repeat the same random behavior. For this purpose, 
+    we repeat the same seed in the psuedo-random generator.
+    """
+
+    DEFAULT_STD_DEV = 0.1
+    DEFAULT_SUCCESS_PROBABILITY = 0.5
+    DEFAULT_NUMBER_OF_TRIALS = 1
+    
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._num_simulations = kwargs.pop("num_simulations", CausalRefuter.DEFAULT_NUM_SIMULATIONS )
+        self._sample_size = kwargs.pop("sample_size", len(self._data))
+        required_variables = kwargs.pop("required_variables", None)
+        self._noise = kwargs.pop("noise", BootstrapRefuter.DEFAULT_STD_DEV )
+        self._probability_of_change = kwargs.pop("probability_of_change", None)
+        self._random_state = kwargs.pop("random_state", None)
+        self._invert = None
+        # Concatenate the confounders, instruments and effect modifiers
+        self._variables_of_interest = self._target_estimand.backdoor_variables + \
+                                      self._target_estimand.instrumental_variables + \
+                                      self._estimate.params['effect_modifiers']
+
+        if 'logging_level' in kwargs:
+            logging.basicConfig(level=kwargs['logging_level'])
+        else:
+            logging.basicConfig(level=logging.INFO)
+        self.logger = logging.getLogger(__name__)
+        
+        # Sanity check the parameters passed by the user
+        # If the data is invalid, we throw the corresponding error
+        if required_variables is int:
+            if len(self._variables_of_interest) < required_variables:
+                self.logger.error("Too many variables passed.\n The number of  variables is: {}.\n The number of variables passed: {}".format(
+                    len(self._variables_of_interest),
+                    required_variables )
+                )
+                raise ValueError("The number of variables in the arguments is greater than the number of variables")
```

Comment:
which arguments? better to say the name of parameter that user must have passed.

I.e., "The number of variables passed by required_variables is greater than the number of variables in the dataset (excluding treatment and outcome)"